src/MBP_UserImport_Source_AfterSchool.php

```diff
@@ -19,16 +19,83 @@ class MBP_UserImport_Source_AfterSchool extends MBP_UserImport_BaseSource
   /**
    * Supported key / columns in CSV file from source.
    */
-  private function setKeys() {
+  protected function setKeys() {
 
     $keys = [
-      '???',
+      'SentToPhone',
+      'SenderName',
+      'ReceiverName',
+      'SchoolID',
+      'SchoolName',
+      'SchoolShort',
+      'SchoolAbbreviation',
+      'Message'
     ];
 
     return $keys;
   }
   
   /**
+   * Logic to determine if data from row in After School CSV file can be processed.
+   *
+   * @param array $data
+   *   Values being processed from row of data in CSV file.
+   */
+  public function canProcess($data) {
+
+    if (empty($data['SentToPhone'])) {
+      return false;
+    }
+
+    // Validate phone number based on the North American Numbering Plan
+    // https://en.wikipedia.org/wiki/North_American_Numbering_Plan
+    $regex = "/^(\d[\s-]?)?[\(\[\s-]{0,2}?\d{3}[\)\]\s-]{0,2}?\d{3}[\s-]?\d{4}$/i";
+    $mobile = str_replace("'", '', $data['SentToPhone']);
+    if (!(preg_match( $regex, $mobile))) {
+      echo '** canProcess(): Invalid phone number based on North American Numbering Plan standard: ' .  $mobile, PHP_EOL;
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Assign columns specific to the After School CSV file to common columns expected
+   * by the consumer.
+   *
+   * @param array $data
+   *   Values collected from the CSV file for assignment to expected indexes in the consumer.
+   */
+  public function setter(&$data) {
+
+    $message = [];
+    $message['mobile'] = (int) str_replace("'",'', $data['SentToPhone']);
+
+    $data['SenderName'] = str_replace("'",'', $data['SenderName']);
+    $nameBits = explode(' ',$data['SenderName']);
+    if (count($nameBits) > 1) {
+      $message['last_name'] = array_pop($nameBits);
+      $message['first_name'] = implode(' ', $nameBits);
+    }
+    else {
+      $message['first_name'] = $nameBits[0];
+    }
+
+    $message['hs_name'] = str_replace("'",'', $data['SchoolShort']);
```

Comment:
The message sent between the producer (this code) and the consumer (mbc-user-import) is in JSON. Without removing the "'" the JSON will look like:
```
"hs_name":"'DoSomething'",
```
The extra "'" look messy to me.